models/asymkey/display_key.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2025 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package asymkey
+
+import (
+	"os"
+
+	"code.gitea.io/gitea/modules/git"
+)
+
+func GetDisplaySigningKey(key *git.SigningKey) (string, error) {
```

Comment:
The struct is sufficiently clear and readable, so we should use it directly.
```
type SigningKey struct {
	KeyID  string
	Format string
}
```
The issue is that KeyID should represent the fingerprint, not the file path of the SSH key. We probably shouldn’t be storing the SSH file path in the KeyID field.

Reply:
No, it is insufficient for other usages.

You will make a mess by " We probably shouldn’t be storing the SSH file path in the KeyID field."

Reply:
> I have been trapped by the dynamic connection between code and tmpl files, that ide intellisense do not understand

Will take a look